blueprint/core/blueprint_ShadowView.cpp

```diff
@@ -170,122 +309,157 @@ namespace blueprint
+            return;
+        }
+        if (name == flexProperties.flexGrow)
+        {
             BP_SET_FLEX_FLOAT_PROPERTY(newValue, YGNodeStyleSetFlexGrow, yogaNode)
-        if (name == juce::Identifier("flex-shrink"))
+            return;
+        }
+        if (name == flexProperties.flexShrink)
+        {
             BP_SET_FLEX_FLOAT_PROPERTY(newValue, YGNodeStyleSetFlexShrink, yogaNode)
-        if (name == juce::Identifier("flex-basis"))
+            return;
+        }
+        if (name == flexProperties.flexBasis)
+        {
             BP_SET_FLEX_DIMENSION_PROPERTY_AUTO(newValue, YGNodeStyleSetFlexBasis, yogaNode)
-        if (name == juce::Identifier("width"))
+            return;
+        }
+        if (name == flexProperties.width)
+        {
             BP_SET_FLEX_DIMENSION_PROPERTY_AUTO(newValue, YGNodeStyleSetWidth, yogaNode)
-        if (name == juce::Identifier("height"))
+            return;
+        }
+        if (name == flexProperties.height)
+        {
             BP_SET_FLEX_DIMENSION_PROPERTY_AUTO(newValue, YGNodeStyleSetHeight, yogaNode)
-        if (name == juce::Identifier("min-width"))
+            return;
+        }
+        if (name == flexProperties.minWidth)
+        {
             BP_SET_FLEX_DIMENSION_PROPERTY(newValue, YGNodeStyleSetMinWidth, yogaNode)
-        if (name == juce::Identifier("min-height"))
+            return;
+        }
+        if (name == flexProperties.minHeight)
+        {
             BP_SET_FLEX_DIMENSION_PROPERTY(newValue, YGNodeStyleSetMinHeight, yogaNode)
-        if (name == juce::Identifier("max-width"))
+            return;
+        }
+        if (name == flexProperties.maxWidth)
+        {
             BP_SET_FLEX_DIMENSION_PROPERTY(newValue, YGNodeStyleSetMaxWidth, yogaNode)
-        if (name == juce::Identifier("max-height"))
+            return;
+        }
+        if (name == flexProperties.maxHeight)
+        {
             BP_SET_FLEX_DIMENSION_PROPERTY(newValue, YGNodeStyleSetMaxHeight, yogaNode)
-        if (name == juce::Identifier("aspect-ratio"))
+            return;
+        }
+        if (name == flexProperties.aspectRatio)
+        {
             BP_SET_FLEX_FLOAT_PROPERTY(newValue, YGNodeStyleSetAspectRatio, yogaNode)
+            return;
+        }
 
         //==============================================================================
         // Margin
-        juce::Identifier marginMetaProp ("margin");
-
-        for (const auto& [edgeName, enumValue] : ValidEdgeValues)
+        if (isMarginProp(name))
         {
-            juce::Identifier propId (juce::String("margin-") + edgeName);
+            if (name == flexProperties.marginMetaProp)
+                BP_SET_FLEX_DIMENSION_PROPERTY_AUTO(newValue, YGNodeStyleSetMargin, yogaNode, YGEdgeAll)
+            else
+                BP_SET_FLEX_DIMENSION_PROPERTY_AUTO(newValue, YGNodeStyleSetMargin, yogaNode, ValidEdgeValues[name.toString().replace("margin-", "")])
```

Comment:
OK with this removal of the  "margin-" prefix here?

Reply:
Pretty sure we don't need that `toString().replace` at all? `name == flexProperties.marginMetaProp` which doesn't include a `"margin-"`. Or I'm missing something :)